plexus-compilers/plexus-compiler-javac/src/main/java/org/codehaus/plexus/compiler/javac/JavacCompiler.java

```diff
@@ -128,6 +133,39 @@ public String getCompilerId() {
         return "javac";
     }
 
+    private String getInProcessJavacVersion() throws CompilerException {
+        return System.getProperty("java.version");
+    }
+
+    private String getOutOfProcessJavacVersion(String executable) throws CompilerException {
```

Comment:
I would like to consider caching an output of additional external executable. 
It can take a notable time especially in multi module projects.

We can simply build a static map executable to version.

Reply:
Fixed in https://github.com/codehaus-plexus/plexus-compiler/pull/357/commits/37195795d24d0e6a09e493b65d1901233329ba98.